src/points.jl

```diff
@@ -22,6 +22,7 @@ O = Point(0.0, 0.0) # origin of 2D Euclidean space
 struct Point{Dim,T}
   coords::SVector{Dim,T}
   Point{Dim,T}(coords::SVector) where {Dim,T} = new{Dim,T}(coords)
+  Point{Dim,T}(coords::SVector{Dim,T2}) where {Dim,T,T2<:Integer} = new{Dim,Float64}(coords)
```

Comment:
Maybe we need to convert the type differently?

```julia
struct Point{Dim,T}
  coords::SVector{Dim,T}
  function Point{Dim,T}(coords) where {Dim,T}
    eltype(coords) <: Integer ? new(Float64.(coords)) : new(coords)
  end
end
```

Does that work? I am not very familiar with the machinery behind inner constructors. We also need to double check that this is efficient.

Reply:
Maybe we should use the multiple-dispatch to dispatch `SVector{Dim,T} where T<:Integer` separately instead of make it manual. I am not familiar that, too.

Reply:
I am not sure we need inner constructors at all. Maybe we can keep the default constructors, and add a method `Point(coords::SVector{Dim,<:Integer}) where {Dim} = Point(convert(SVector{Dim,Float64}, coords))`

Reply:
Yes, I find inner constructors a bit difficult to grasp, but remember someone added it to handle the ambiguities with the outer constructors somehow. If we can eliminate the inner constructor and still convert Integer coordinates do Float64 that would be a double win. Appreciate any help on this.

I think our test suite is already covering the past issues people had with Point constructors.

Reply:
@HaoxuanGuo did your previous attempt with an external constructor worked fine? All tests passed?

Reply:
@juliohm Both the 2 inner or outer one works well. You may notice that there are some test cases not passed. Because they are

```julia
@test coordtype(Point(1, 1)) == Int
@test coordtype(Point([1])) == Int
@test coordtype(Point([1,2])) == Int
@test coordtype(Point([1,2,3])) == Int
```

Obviously, we want to convert them to `Float64` instead of `Int`. It not passed is just our expected.

So, both them are worked fine, but not all tests passed. While **`not all tests passed` is our expected**.

Reply:
Nice, feel free to update the tests to match the new behavior. I can review the PR again by the end of the day. Thanks for this must needed addition.